cmd/kube-apiserver/app/server.go

```diff
@@ -133,7 +133,10 @@ func Run(s *options.APIServer) error {
 	}
 
 	// Proxying to pods and services is IP-based... don't expect to be able to verify the hostname
-	proxyTLSClientConfig := &tls.Config{InsecureSkipVerify: true}
+	proxyTLSClientConfig, err := transport.TLSConfigFor(s.KubeletConfig.TransportConfig())
```

Comment:
Both the kubelet client and the proxy transport are passed to things like pod storage:
https://github.com/kayrus/kubernetes/blob/ebbcb11705ce40b3afb3cdcf7542f055444adfb7/pkg/master/master.go#L352

The kubelet client is used for logs/attach/exec calls